@ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx

```diff
@@ -0,0 +1,203 @@
+    fireEvent.click(datePickerElement)
+    expect(datePickerElement).toBeEnabled()
+  })
+
+  it('click on label focusing date picker', () => {
+    render(
+      <div>
+        <label htmlFor="id" id="label">
+          label
+        </label>
+        <MainDatePicker id="id" />
+      </div>
+    )
+
+    const labelElement = screen.getByText('label')
+    fireEvent.click(labelElement)
+    expect(screen.getByPlaceholderText('MM/DD/YYYY')).toBeEnabled()
+  })
+
+  it('clicking on the "number" button changes the date', () => {
+    render(<MainDatePicker value="10/10/2000" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    fireEvent.click(screen.getByRole('gridcell', { name: '24' }))
+    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('10/24/2000')
+  })
+
+  it('clicking on the "next month" button changes the month', async () => {
+    render(<MainDatePicker value="10/10/2000" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    fireEvent.click(screen.getByTitle('Next month'))
+    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
+    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('11/20/2000')
+  })
+
+  it('clicking on the "previous month" button changes the month', async () => {
+    render(<MainDatePicker value="10/10/2000" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    fireEvent.click(screen.getByTitle('Previous month'))
+    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
+    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('09/20/2000')
+  })
+
+  it('clicking on the "next month" button on last month changes year', async () => {
+    render(<MainDatePicker value="12/10/2000" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    fireEvent.click(screen.getByTitle('Next month'))
+    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
+    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('01/20/2001')
+  })
+
+  it('clicking on the "previous month" button on first month changes year', async () => {
+    render(<MainDatePicker value="01/10/2000" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    fireEvent.click(screen.getByTitle('Previous month'))
+    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
+    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('12/20/1999')
+  })
+
+  it('"Previous month" button is disabled on first month of first year', async () => {
+    render(<MainDatePicker value="01/01/1900" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    expect(screen.getByTitle('Previous month')).toBeDisabled()
+  })
+
+  it('"Next month" button is disabled on last month of last year', async () => {
+    render(<MainDatePicker value="12/31/2099" />)
+
+    fireEvent.click(screen.getByRole('button'))
```

Comment:
screen.getByRole('button') 
you can use variables , try not to repeat yourself

Reply:
done